src/OpenTelemetry.Instrumentation.AWSLambda/Implementation/AWSLambdaHttpUtil.cs

```diff
@@ -0,0 +1,147 @@
+// <copyright file="AWSLambdaHttpUtil.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Web;
+using Amazon.Lambda.APIGatewayEvents;
+using OpenTelemetry.Trace;
+
+namespace OpenTelemetry.Instrumentation.AWSLambda.Implementation
+{
+    internal class AWSLambdaHttpUtil
+    {
+        // x-forwarded-... headres are described here https://docs.aws.amazon.com/elasticloadbalancing/latest/classic/x-forwarded-headers.html
+        private const string HeaderXForwardedProto = "x-forwarded-proto";
+        private const string HeaderHost = "host";
+
+        internal static IEnumerable<KeyValuePair<string, object>> GetHttpTags<TInput>(TInput input)
+        {
+            var tags = new List<KeyValuePair<string, object>>();
+
+            string httpScheme = null;
+            string httpTarget = null;
+            string httpMethod = null;
+            string hostName = null;
+            int? hostPort = null;
+
+            switch (input)
+            {
+                case APIGatewayProxyRequest request:
+                    httpScheme = GetHeaderValue(request, HeaderXForwardedProto);
+                    httpTarget = string.Concat(request.Path ?? string.Empty, ConstructQueryString(request.QueryStringParameters));
+                    httpMethod = request.HttpMethod;
+                    var hostHeader = GetHeaderValue(request, HeaderHost);
+                    (hostName, hostPort) = GetHostAndPort(httpScheme, hostHeader);
+                    break;
+                case APIGatewayHttpApiV2ProxyRequest requestV2:
+                    httpScheme = GetHeaderValue(requestV2, HeaderXForwardedProto);
+                    httpTarget = string.Concat(requestV2?.RequestContext?.Http?.Path ?? string.Empty, ConstructQueryString(requestV2.QueryStringParameters));
```

Comment:
Let's use the `rawQueryString` property of the V2 request instead.

Reply:
The new implementation uses the `RawQueryString` property as is and appends it to the `RawPath` after `?`.